<!DOCTYPE html>
<html lang="en-uk">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.24.1" />
  <meta name="author" content="Bioinformatics Group">
  

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.0/css/academicons.min.css" integrity="sha512-GGGNUPDhnG8LEAEDsjqYIQns+Gu8RBs4j5XGlxl7UfRaZBhCCm5jenJkeJL8uPuOXGqgl8/H1gjlWQDRjd3cUQ==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-101891200-1', 'auto');
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  <link rel="alternate" href="/index.xml" type="application/rss+xml" title="UoP Bioinformatics Group">
  <link rel="feed" href="/index.xml" type="application/rss+xml" title="UoP Bioinformatics Group">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="/talk/2017-11-07_introduction-to-bioinformatics/">

  

  <title>Introduction to Bioinformatics | UoP Bioinformatics Group</title>

</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">UoP Bioinformatics Group</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#resources">
            
            <span>Resources</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#software">
            
            <span>Software</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#talks">
            
            <span>Talks</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>

<div class="pub" itemscope itemtype="http://schema.org/CreativeWork">

  
<div class="article-header">
  <img src="/img/headers/bubbles-wide.jpg" class="article-banner" itemprop="image">
  
</div>



  <div class="article-container">

    <div class="pub-title">
      <h1 itemprop="name">Introduction to Bioinformatics</h1>
      <span class="pub-authors" itemprop="author">
        
      </span>
      <span class="pull-right">
        
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2ftalk%2f2017-11-07_introduction-to-bioinformatics%2f"
         target="_blank">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=Introduction%20to%20Bioinformatics&amp;url=%2ftalk%2f2017-11-07_introduction-to-bioinformatics%2f"
         target="_blank">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2ftalk%2f2017-11-07_introduction-to-bioinformatics%2f&amp;title=Introduction%20to%20Bioinformatics"
         target="_blank">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2ftalk%2f2017-11-07_introduction-to-bioinformatics%2f&amp;title=Introduction%20to%20Bioinformatics"
         target="_blank">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=Introduction%20to%20Bioinformatics&amp;body=%2ftalk%2f2017-11-07_introduction-to-bioinformatics%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


      </span>
    </div>

    
      <h3>Abstract</h3>
      <p class="pub-abstract" itemprop="text">An introductory lecture giving a brief outline of what bioinformatics entails, the role of bioinformatics in modern biological sciences research, and the tools that a bioinformatician must learn to analyse the large body of data generated from such research. This will be a 2 hour introductory session geared towards MSc students in the School of Pharmacy and Biomedical Sciences.</p>
    

    <div class="row">
      <div class="col-sm-1"></div>
      <div class="col-sm-10">
        <div class="row">
          <div class="col-xs-12 col-sm-3 pub-row-heading">Date</div>
          <div class="col-xs-12 col-sm-9" itemprop="datePublished">
            Tue, Nov 7, 2017
          </div>
        </div>
      </div>
      <div class="col-sm-1"></div>
    </div>
    <div class="visible-xs space-below"></div>

    <div class="row">
      <div class="col-sm-1"></div>
      <div class="col-sm-10">
        <div class="row">
          <div class="col-xs-12 col-sm-3 pub-row-heading">Event</div>
          <div class="col-xs-12 col-sm-9">
            <a href="http://www.port.ac.uk/courses/health-sciences-and-social-work/msc-medical-biotechnology/" target="_blank">
            MSc Medical Biotechnology
            </a>
          </div>
        </div>
      </div>
      <div class="col-sm-1"></div>
    </div>
    <div class="visible-xs space-below"></div>

    
    <div class="row">
      <div class="col-sm-1"></div>
      <div class="col-sm-10">
        <div class="row">
          <div class="col-xs-12 col-sm-3 pub-row-heading">Location</div>
          <div class="col-xs-12 col-sm-9">Portsmouth, United Kingdom</div>
        </div>
      </div>
      <div class="col-sm-1"></div>
    </div>
    <div class="visible-xs space-below"></div>
    

    

    <div class="space-below"></div>

    <div class="article-style">
      

<h2 id="introduction">Introduction</h2>

<p>The term &ldquo;Bioinformatics&rdquo; can apply to a wide range of disciplines in the biological sciences, and the field lies at the interface between biology, statistics, mathematics and computer science. The following are suggested training modules that represent what I believe to be the key aspects of Bioinformatics knowledge essential for anyone looking to analyse next generation sequencing (NGS) data. In this first lecture, we will discuss the following topics in brief to gain an understanding of what Bioinformatics actually is and how essential it is becoming as Big Data becomes the norm in the Biological Sciences.</p>

<h2 id="bioinformatics-essentials">Bioinformatics Essentials</h2>

<p>This series of modules aims to cover some of the basics of Bioinformatics.</p>

<h3 id="1-what-is-bioinformatics">1) What is Bioinformatics?</h3>

<p>This course will introduce the basic concepts behind Bioinformatics analyses, and the reasons why Bioinformaticians are becoming more and more essential to cope with the influx of data as technology continues to improve. We will discuss some examples of how computational methods can help to solve biological problems, and the sorts of typical analyses that a Bioinformatician may expect to conduct.</p>

<h3 id="2-sequence-alignment">2) Sequence Alignment</h3>

<p>This course will focus on the idea of sequence alignments, aiming to find the &ldquo;best&rdquo; alignment between two sequences. In Bioinformatics, we often want to identify how similar two sequences are, be these sequences of nucleotides in DNA and RNA, or sequences of amino acids in proteins. The concept of a &ldquo;best&rdquo; alignment will be discussed, as often how we interpret this will be dependent on the type of data and the questions that we are hoping to answer.</p>

<p>We will also discuss some of the biology behind these questions, why we might be interested in the similarity between two sequences, and how these may relate to the evolution of functional elements between species.</p>

<p>We will discuss some of the algorithms available for performing alignments (e.g. dynamic programming methods such as Smith-Waterman alignments, or heuristic k-tuple methods such as those utilised by BLAST and BLAT) and the various parameters that need to be considered for such analyses (e.g. gap penalties, substitution matrix selection, etc.).</p>

<p>We will touch on the FASTA format of representing sequences, since these are typically the format of choice for most alignment tools, and will discuss some of the more well known sequence alignment tools such as ClustalW for alignment of multiple sequences (e.g. in homology searches) and BLAST for sequence pattern searches in protein and nucleotide databases (such as Ensembl).</p>

<p>We will also discuss how this leads into motif analyses; searching for common short patterns shared between multiple sequences. Such motifs may represent common DNA binding sites (e.g. transcription factor binding sites), or functional protein domains with shared functions.</p>

<p>Finally, this will lead into the idea of pattern matching of short sequences against a complete genome, which is the basis of NGS.</p>

<h3 id="3-interrogation-of-bioinformatics-resources">3) Interrogation of Bioinformatics Resources</h3>

<p>Often Bioinformaticians will need to access published data or data curated by some external resource to compare against their own data. In this module we will discuss some of the more common databases (GenBank, RefSeq, UCSC, Uniprot, Ensembl, ArrayExpress, GEO, SRA, Pfam, Interpro, MiRBase, KEGG, etc). This course will discuss the role of some of these databases, how they are maintained and who maintains them, what data can be accessed from them, and how to access the data. We will also discuss how to submit data to curated databases such as ArrayExpress and GEO, which is typically a requirement of publication of any kind of sequencing based project.</p>

<h3 id="4-the-genome-browser">4) The Genome Browser</h3>

<p>This course will look at how to visualise genomic data using genome browsers such as the University of California, Santa Cruz (UCSC) genome browser or the Integrative Genomics Viewer (IGV). The concepts of data tracks will be discussed, including coverage tracks and bed tracks. We will discuss some of the tracks available in the UCSC browser including annotation from various databases, conservation, and links to the incredibly useful ENCODE (Encylopedia of DNA Elements) data set. We will also look at how we can upload our own tracks, how we can use the table browser to access data, how to convert between different genome builds using the liftOver tool, and how we can set up and share sessions.</p>

<h2 id="next-generation-sequencing">Next Generation Sequencing</h2>

<p>The following modules aim to study the theories underlying NGS analyses, the methods that are available to researchers and how they can be used, and the tools available for the analysis of these data.</p>

<h3 id="1-introduction">1) Introduction</h3>

<p>This module will aim to introduce the concept of NGS, and the broad range of analyses that it opens up to researchers. The idea behind NGS is to utilise the power of modern sequencing chemistries to interrogate a large number of short nucleotide fragments simultaneously for global analyses. We will discuss the evolution of sequencing from the Sanger sequencing techniques used to build the first human genome, to the current Illumina sequencing-by-synthesis approach that allows parallel reading of millions of sequence fragments in a single run.</p>

<p>We will discuss how the current sequencing technologies work (focusing primarily on Illumina, but also discussing 454 pyrosequencing, SOLiD, Ion Torrent, etc.), and some of the key considerations for the generation of libraries to be used in sequencing reactions. We will focus mostly on Illumina technologies and will discuss the library preparation methods, how bridge amplification works to generate clusters on the flow cell, the cyclic nature of fluorescently labeled nucleotide incorporation to generate sequence reads, and how base calling scores are generated by the machine.</p>

<p>The idea of NGS is to either resequence an unknown genome to try and build contigs into a complete scaffold, or else to map short fragments against a known genome build in order to quantify different loci (e.g. regions of DNA bound by a transcription factor, or expressed genes). We will discuss the idea of read &ldquo;mapping&rdquo;, and the different algorithms available for this.</p>

<p>In particular we will focus on two main NGS applications:</p>

<ol>
<li><strong>ChIP-seq</strong>: Sequencing of DNA fragments identified through immunoprecipitation pulldown, typically used to identify protein-DNA interactions (e.g. transcription factor binding) or DNA bound to histones showing specific modifications (e.g. H3K4me3)</li>
<li><strong>RNA-seq</strong>: Sequencing of RNA fragments in order to profile gene (and other non-coding transcript) expression</li>
</ol>

<p>We will discuss the key differences between these two types of data (in particular the difference between mapping to the genome and mapping to the transcriptome), and what this means for mapping. We will contrast genomic mapping algorithms (e.g. BWA and Bowtie) with splice-aware mapping algorithms (e.g. Tophat2 and STAR).</p>

<p>We will briefly discuss the broad range of additional sequencing approaches available to researchers (RIP-seq, CLIP-seq, ChIRP-seq, Bisulphite-seq, NET-seq, etc.), and the situations in which these might be of use.</p>

<h3 id="2-experimental-design">2) Experimental Design</h3>

<p>This module will focus more on the types of considerations that must be undertaken by a Bioinformatician when designing an NGS experiment. Along with more basic elements of experimental design such as power calculations, determining the minimum number of replicates, blocking and randomisation to remove systematic bias, etc, we will focus on how one goes about selecting the optimum library design for the question of interest. This includes whether or not paired end sequencing should be considered, the number of sequencing cycles to use, whether to use multiplexing, the read depth required and the number of lanes required to achieve this, whether a stranded protocol should be utilised, etc.</p>

<p>In addition, certain NGS specific issues will be discussed such as selection of multiplex barcodes to ensure sequence complexity is maintained, balancing of multiplexed samples to avoid biasing a sequencing run for a single sample, how best to mutliplex samples over multiple lanes, adapter contamination and how to avoid it, PCR bias and how to avoid it, selection of control samples (e.g. input controls and IgG negative controls), etc.</p>

<p>We will discuss how best to avoid typical pitfalls of NGS experiments, and how to deal with issues when they do arise. We will also discuss some of the issues inherent to different sequencing technologies, such as how best to account for ribosomal contamination in total RNA when looking at RNA seq (ribozero treatment, polyA selection, or pre-mapping against a ribosomal transcriptome), or how to deal with cytosine-depletion in bisulphite treated sequencing samples.</p>

<h3 id="3-mapping">3) Mapping</h3>

<p>This module will focus on what to do with the raw data that comes from an NGS experiment. We will discuss the fastq file format, and how the base quality scoring is represented in Phred format. We will touch on the fact that the encoding for these can be different depending on when and where the sequencing was performed.</p>

<p>We will discuss the initial quality control steps that should be performed to determine the quality of these data prior to mapping. For instance we can check that the reads do indeed come from the intended species with no contamination, we can check for the presence of adapter contamination (or any other sequence bias) that may represent a library problem that needs to be resolved prior to mapping, we can check that base composition is roughly uniform across the read set, we can check for read duplication which may represent PCR bias in the library preparation stage, etc.</p>

<p>We will discuss the idea of read trimming and some of the available tools for this (trim-galore, trimmomatic, fastx toolkit), to remove low-quality bases and contaminating sequences which may prevent accurate mapping.</p>

<p>We will discuss how to go about choosing the best mapping algorithm for your needs and how to tune the parameters to get the most accurate mapping from your data. The idea of mapping quality will be discussed, and how this is determined. In particular we will discuss issues that can cause ambiguity in mapping. For instance we may see many reads mapping to a single location, which may represent a result of PCR amplification bias. Or we may see a single read map to many locations (multi-reads), typically representing duplicated regions of the genome (e.g. segmental duplications, repetitive elements and the like). There is currently no consensus on how best to deal with these ambiguities, but knowing the options is essential.</p>

<p>We will talk about the current standardised format for mapped data (SAM and BAM format), and how to understand the meaning of the different fields. We will use the samtools suite of tools for interrogating, sorting, filtering, etc. these files, as well as additional tools for generating formats for use with genome browsers such as UCSC and IGV (e.g. the bedtools suite which can convert SAM/BAM format files into genome coverage files such as bedgraph or wiggle files).</p>

<h3 id="4-next-generation-sequencing-analysis">4) Next Generation Sequencing Analysis</h3>

<p>This module will focus on what to do with our mapped data, and will focus primarily on the analysis of ChIP-seq and RNA-seq data. For ChIP-seq, the idea is to identify loci that are enriched for the factor that we are interrogating. For many factors (e.g. transcription factors) this comes down to identifying &ldquo;peaks&rdquo; in the data that are significantly enriched above the background. Various peak calling algorithms are available for this, and we will discuss some of these (e.g. MACS, FindPeaks, SICER, etc.) and how they work. Some factors (e.g. histone modifications such as H3K36me3) are not represented by clearly defined peaks, but rather by more protracted regions of enrichment, and so different approaches may need to be considered for these. We will discuss the kinds of analyses typical to ChIP-seq, such as annotating the location of peaks relative to known features (e.g. gene TSS, promoters, exons, introns, enhancers, superenhancers, etc.), comparing profile plots centered at particular regions (e.g. centering all samples at the peaks called for one of the samples to look for correlation, or centering at the TSS to look at promoter association), or identifying differentially regulated regions between conditions.</p>

<p>For RNA-seq, the approach is very different, as we aim to use the mapped reads to generate a quantitative count for known transcripts in order to compare and contrast expression levels between conditions. The idea is to count up reads within the coding region of known genes, although this brings up a number of issues that need to be considered. To avoid biasing our data, we need to ensure that each read is assigned to only a single transcript, but often genes have multiple transcipts consisting of different combinations of exons. Determining which of the transcripts a particular read belongs can be an incredibly tricky endeavor. Also, we see overlapping transcripts in the genome, or features within features (e.g. ncRNAs in introns) which increase this ambiguity. It is very important to be aware of these issues when trying to quantify your transcripts. Typically analyses utilise these per-transcript or per-gene counts and look for differential expression between conditions to find significantly changing candidates. These genes can then be further analysed by looking for common functions (e.g. through the use of gene ontology analyses), and by comparing between different conditions (e.g. different time points) to determine a network of gene expression changes.</p>

<h2 id="programming">Programming</h2>

<h3 id="1-thinking-like-a-programmer">1) Thinking like a programmer</h3>

<p>The concept of writing computer code can be a daunting prospect, particularly for people who have had no experience with the basic concepts behind seemingly complex code syntax. This module will be a basic course, introducing people with no computing experience to the concepts of writing programs, mainly through the use of pseudo-code.</p>

<p>Like spoken languages, the concepts underlying most programming languages are very similar. This course will focus on the underlying &ldquo;grammar&rdquo; of programming, and not on any one specific language. This includes understanding concepts such as variables, data classes, subroutines/functions, input/output, iteration using loops, conditional statements, boolean operators, comparison operators, basic logic, and best practices for writing code.</p>

<p>With these tools, students will be much better equipped to understand how to answer biological questions using programmatic means.</p>

<h3 id="2-data-processing-with-perl">2) Data processing with perl</h3>

<p>In Bioinformatics, we deal with large amounts of data and often need to find ways to interrogate, summarize, modify and convert them. Perl is a highly efficient language for such tasks, and this module will focus on how the perl language can be used to accomplish these tasks.</p>

<p>We will first look at the basic structure of a perl program, and the specific syntax for this language. The course will start with an introduction to the basic programming concepts discussed in the previous module, as well as more esoteric perl-specific concepts. We will discuss the differences between scalar, array, and hash variables, and will also delve into the concept of variable referencing. We will discuss the concept of streams and file handles for reading and writing data, and how to deal with different types of data that you may encounter. We will look at how to install new perl modules to add new functionality depending on your goals. We will look at the idea of scripting, and creating programs that can be run with user input to achieve a specific task (e.g. writing a script to pull out a certain number of sequences from a fastq file). We will discuss ways to ensure that our code is efficient for dealing with large amounts of data.</p>

<p>We will also go into some detail about the concept of regular expressions for pattern recognition, which can be incredibly important concept in dealing with DNA, RNA and protein sequences. We will finally discuss using perl in real world situations, playing with some example data files to get a feel for how these concepts will work in practice.</p>

<h3 id="3-data-analysis-with-r">3) Data analysis with R</h3>

<p>R is an open source statistical language, used in many data sciences for statistical analyses. R libraries cover most statistical algorithms and allow for plotting of publication-quality figures. It is very well documented with a simple syntax, and is an excellent tool for data analysis.</p>

<p>In this module, we will first begin to understand the basic syntax of R, and how to load in and analyse data. In particular, the data frame class and how to interrogate these for data analysis will be considered.</p>

<p>We will also cover some statistics in this module, discussing the role of hypothesis testing in science and how we can go about setting up models to test our theories. Starting with basic summary statistics (such as mean, median, standard deviation, etc), we will eventually delve into the concept of linear models, and how these can be used to test association of potential explanatory variables with the dependent variable. We will look at how this relates to correlation analysis, move into t-test and ANOVA tests for comparing means between groups, non-parametric tests and when to use them, and eventually into more complicated multi-variable models and model selection methods.</p>

<p>We will use example data sets to explore R and some of the powerful analysis techniques that can be used to explore the data.</p>

<h3 id="4-solving-biological-problems-with-r">4) Solving biological problems with R</h3>

<p>This course will extend the previous R course to look specifically at some of the ways that R can be used in biological data analyses. In particular, this course will look at some of the packages available through the Bioconductor project for accomplishing specific tasks that Bioinformaticians frequently come across.</p>

<p>First we will look at some of the more important biological classes in R. For instance, the GRanges class which extends the data range class for defining genomic regions. Understanding how GRanges can be compared and contrasted can make future analysis of any kind of genomic regions in R much easier and more efficient.</p>

<p>We will look at some example RNA seq data, and discuss ways of accessing gene annotation from Biomart and Ensembl, then summarising read level data on a per gene basis. With this summarised data, we will use packages limma, DESeq2 and edgeR to look for differentially expressed genes. We will then explore the various annotation packages available for converting between different accessions (Refseq, Ensembl, Hugo, etc), and move on to the use of annotation packages like GO.db for gene ontology analyses.</p>

<p>Next we will look at sequence packages such as the BioStrings genome packages, and ways to search for sequences to identify motifs, and compare known motifs and consensus sequences against a set of targets.</p>

<p>We will also look at typical visualisation approaches for biological data, particularly sequencing data. For instance, for ChIP seq of transcription factors, we may wish to look at how the binding looks at the TSS of known gene. For this, we can count reads in windows around known TSS, and use heatmaps to visualise the enrichment.</p>

<p>In addition we will look at ways to import and export data using the rtracklayer package, and how to port track data for visualisation using a genome browser (such as UCSC or IGV).</p>

<p>Finally we will touch upon the use of R Markdown, a way of creating dynamic documents with embedded R analyses which can be used to generate reports. Using <em>knitr</em> in R, R Markdown files containing embedded R code can be converted to documents of various formats (pdf, Microsoft Word, slideshow presentations, HTML web pages, and even interactive web applications using the <em>shiny</em> library) containing data, figures and tables generated on the fly. In particular, the idea of using R Markdown as a form of lab book will be suggested so that analyses are well documented.</p>

<h3 id="5-version-control">5) Version Control</h3>

<p>One important aspect of programming is version control, a method of managing and logging all changes made to programs in order to track the evolution of the code. Often a piece of code will serve a single purpose, will be used for a particular task, and then never consulted again. However, sometimes you will find yourself reusing, and more importantly <em>changing</em> code for future use. It is important to track and monitor these changes, particularly if you find that you ultimately break your code in unexpected ways.</p>

<p>Version control systems such as Git allow you to track changes as you make them, create branches of your working tree that can later be merged to allow testing of new ideas before committing to them, and tag your work at specific times (e.g. for incremental releases). Git also allows multiple users to work on a single project and allows all users to keep their code in sync with one another, which can be essential when working collaboratively with other Bioinformaticians.</p>

<p>This course will run through some of the basics of Git version control, how to initialise a repository and track changes as you go, how to create, merge and delete multiple branches, and how to fork repositories for collaborative coding.</p>

    </div>

  </div>
</div>

<div class="container">
  <nav>
  <ul class="pager">
    
    <li class="previous"><a href="/talk/2017-06-16_bioinformatics-research-at-the-university-of-portsmouth/"><span
      aria-hidden="true">&larr;</span> Bioinformatics Research at the University of Portsmouth</a></li>
    

    
  </ul>
</nav>

</div>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2017 Sam Robson &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/1.12.4/jquery.min.js" integrity="sha512-jGsMH83oKe9asCpkOVkBnUrDDTp8wl+adkB2D+//JtlxO4SrLoJdhbOysIFQJloQFD+C4Fl1rMsQZF76JjV0eQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.2/imagesloaded.pkgd.min.js" integrity="sha512-iHzEu7GbSc705hE2skyH6/AlTpOfBmkx7nUqTLGzPYR+C1tRaItbRlJ7hT/D3YQ9SV0fqLKzp4XY9wKulTBGTw==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gsap/1.19.1/TweenMax.min.js" integrity="sha512-Z5heTz36xTemt1TbtbfXtTq5lMfYnOkXM2/eWcTTiLU01+Sw4ku1i7vScDc8fWhrP2abz9GQzgKH5NGBLoYlAw==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gsap/1.19.1/plugins/ScrollToPlugin.min.js" integrity="sha512-CDeU7pRtkPX6XJtF/gcFWlEwyaX7mcAp5sO3VIu/ylsdR74wEw4wmBpD5yYTrmMAiAboi9thyBUr1vXRPA7t0Q==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/languages/r.min.js"></script>
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/languages/perl.min.js"></script>
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/languages/yaml.min.js"></script>
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/languages/python.min.js"></script>
      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

